package/MDAnalysis/analysis/msd.py

```diff
@@ -0,0 +1,355 @@
+Now that we have identified what segment of our MSD to analyse, lets compute a self diffusivity.
+
+Computing Self Diffusivity
+--------------------------------
+Self diffusivity is closely related to the MSD.
+
+.. math::
+
+   D_d = \frac{1}{2d} \lim_{t \to \infty} \frac{d}{dt} MSD(r_{d}) 
+
+From the MSD, self diffusivities :math:`D` with the desired dimensionality :math:`d` can be computed by fitting the MSD with respect to the lag time to a linear model. 
+An example of this is shown below, using the MSD computed in the example above. The segment between :math:`\tau = 20` and :math:`\tau = 80` is used to demonstrate selection of an MSD segment.
+
+    >>> from scipy.stats import linregress as lr
+    >>> start_time = 20
+    >>> start_index = start_time/timestep
+    >>> end_time = 80
+    >>> end_index = end_time/timestep
+    >>> linear_model = lr(lagtimes[start_index:end_index], msd[start_index:end_index])
+    >>> slope = linear_model.slope
+    >>> error = linear_model.rvalue
+    >>> D = slope * 1/(2*MSD.dim_fac) #dim_fac is 3 as we computed a 3D msd ('xyz')
+
+We have now computed a self diffusivity!
+
+
+Notes
+_____
+
+There are several factors that must be taken into account when setting up and processing trajectories for computation of self diffusivities.
+These include specific instructions around simulation settings, using unwrapped trajectories and maintaining relativley small elapsed time between saved frames. 
+Additionally corrections for finite size effects are sometimes employed along with varied means of estimating errors.
+The reader is directed to the following review, which describes many of the common pitfalls [Maginn2019]_. There are other ways to compute self diffusivity including from a Green-Kubo integral. At this point in time these methods are beyond the scope of this module
+
+
+References
+----------
+
+.. [Maginn2019] Maginn, E. J.; Messerly, R. A.; Carlson, D. J.; Roe, D. R.; Elliott, J. R. Best Practices for Computing Transport Properties 1. Self-Diffusivity and Viscosity from Equilibrium Molecular Dynamics [Article v1.0]. Living J. Comput. Mol. Sci. 2019, 1 (1).
+.. [Calandri2011] Calandrini, V.; Pellegrini, E.; Calligari, P.; Hinsen, K.; Kneller, G. R. NMoldyn-Interfacing Spectroscopic Experiments, Molecular Dynamics Simulations and Models for Time Correlation Functions. Collect. SFN 2011, 12, 201–232.
+.. [SO2015] https://stackoverflow.com/questions/34222272/computing-mean-square-displacement-using-python-and-fft
+
+Classes and Functions
+---------------------
+
+.. autoclass:: MeanSquaredDisplacement
+    :members:
+
+"""
+
+from __future__ import division, absolute_import
+from six.moves import zip
+
+import os
+import errno
+import warnings
+import bz2
+import functools
+
+import numpy as np
+from numpy.fft import fft,ifft
+import logging
+import MDAnalysis
```

Comment:
did you decide not to use tidynamics for this module?

Reply:
I decided to just onboard the algorithm to avoid future dependency issues / inclusion of extra packages. Current code is tested against tidynamics in the tests to help show correctness, with a view to removing. More than happy to refactor to use tidynamics if you would prefer.

Reply:
now uses tidynamics